### PR TITLE
feat(extension):  implement extensions button and modal

### PR DESCRIPTION
**Pull request by safeamiiir**
- Add extension list and modal in Experiment details page
- Make the Experiment detail component more modular

### CODE REVIEW

File: aqueductcore/frontend/src/types/globalTypes.ts [hunk 1]
```diff
@@ -3,8 +3,11 @@ import {
   ExperimentData,
   ExperimentFile,
   Experiments,
+  PluginInfo,
```
Comment (samiralavi) [inline aqueductcore/frontend/src/types/globalTypes.ts:6]: You are using plugins and extensions at the same time.

**Review by samiralavi — APPROVED** [review]
Looks good to me thanks.

File: aqueductcore/frontend/src/__mocks__/ExtensionsDataMock.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                            "string2",
+                            "string three",
+                            "string4"
+                        ]
+                    }
+                ]
+            }
+        ],
+        "description": "This plugin prints environment variables passed to it. No requests to Aqueduct sent.\n",
+        "authors": "aqueduct@riverlane.com"
```
Comment (jatinriverlane) [inline aqueductcore/frontend/src/__mocks__/ExtensionsDataMock.ts:145]: A minor readability suggestion, these 2 lines can be moved to line 6, similar in the other function.

**Review by jatinriverlane — APPROVED** [review]
Thanks @safeamiiir

**Pull request merged**